grails-app/taglib/org/pih/warehouse/SelectTagLib.groovy

```diff
@@ -501,6 +501,30 @@ class SelectTagLib {
         out << g.select(attrs)
     }
 
+    def selectLocationWithSubmitRequestActivity = { attrs, body ->
+
+        // If attrs.from is populated use that by default even if it's empty
+        if (!attrs.containsKey("from")) {
+            ActivityCode activityCode = attrs.activityCode ?: null
+            attrs.from = locationService.getAllLocations().sort { it?.name?.toLowerCase() }
+
+            // use sparingly - this is expensive since it requires multiple database queries
+            if (activityCode) {
+                attrs.from = attrs.from.findAll { it.supports(activityCode) || (!it.supports(activityCode) && it.supports(ActivityCode.SUBMIT_REQUEST)) }
```

Comment:
This doesn't look right. What are we trying to achieve here and why won't this work by generically passing ActivityCode.SUBMIT_REQUEST as the activityCode to selectLocation?

Basically, I'd like to make it so we only need selectLocation. Is there any way to make that happen?

Reply:
Sure, that is leftover from my trials, sorry for that.